bin/build-locales

```diff
@@ -64,66 +64,82 @@ try {
 
-// get view-specific locale strings.
-var files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localeCompare.getIdsForView(view, file, viewLocales, idsWithICU, icuWithIds);
-});
+// start with all views, and remove localized ones as they are iterated over
+for (var v in routes) {
+    if (typeof routes[v].redirect !== 'undefined') {
+        continue;
+    }
+    
+    views.push(routes[v].name);
+    try {
+        var subdir = routes[v].view.split('/');
+        subdir.pop();
+        var l10n = path.resolve(__dirname, '../src/views/' + subdir.join('/') + '/l10n.json');
+        fs.accessSync(l10n, fs.F_OK);
+        ids = JSON.parse(fs.readFileSync(l10n, 'utf8'));
+        viewLocales[routes[v].name] = {
+            en: ids
+        };
+        idsWithICU = merge(idsWithICU, localeCompare.idToICUMap(routes[v].name, ids));
+        icuWithIds = merge(icuWithIds, localeCompare.icuToIdMap(routes[v].name, ids));
+    } catch (err) {
+        // check that routes' view path is not malformed, error if it is
+        try {
+            fs.accessSync(path.resolve(__dirname, '../src/views/' + routes[v].view + '.jsx'));
+        } catch (err) {
+            // the config for the view is not set up correctly, so throw the error.
+            throw err;
+        }
+    }
 
-// get asset url translations
-var localizedAssetUrls = {};
-files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n-static.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localizedAssetUrls[view] = {};
-
-    var assetUrls = JSON.parse(fs.readFileSync(file, 'utf8'));
-    for (var lang in localizedUrls) {
-        localizedAssetUrls[view][lang] = {};
-        for (var key in assetUrls) {
-            if (localizedUrls[lang].hasOwnProperty(key)) {
-                localizedAssetUrls[view][lang][key] = localizedUrls[lang][key];
-            } else {
-                localizedAssetUrls[view][lang][key] = assetUrls[key];
+    // get asset url translations
+    try {
+        var l10nStatic = path.resolve(__dirname, '../src/views/' + routes[v].view + '/l10n-static.json');
+        fs.accessSync(l10nStatic, fs.F_OK);
+        localizedAssetUrls[routes[v].name] = {};
+
+        var assetUrls = JSON.parse(fs.readFileSync(l10nStatic, 'utf8'));
+        for (var lang in localizedUrls) {
+            localizedAssetUrls[routes[v].name][lang] = {};
+            for (var key in assetUrls) {
+                if (localizedUrls[lang].hasOwnProperty(key)) {
+                    localizedAssetUrls[routes[v].name][lang][key] = localizedUrls[lang][key];
+                } else {
+                    localizedAssetUrls[routes[v].name][lang][key] = assetUrls[key];
+                }
             }
         }
+    } catch (err) {
+        // :)
```

Comment:
Ah whoops – forgot to add to this what I did to the other `l10n` file above. Thanks!